File: tennis-intelligence-platform-web/src/types/fullMatchList.ts
// types/fullMatchList.ts — matches api/schemas/full_match_list.py exactly.

export interface FullMatchSummaryRow {
  match_id: string;
  has_replay_data: boolean;
  tournament: string;
  year: number | null;
  surface: string;
  round: string;
  winner: string;
  loser: string;
  final_score: string | null;
  tournament_level: string | null;
  best_of: number | null;
  winner_elo: number | null;
  loser_elo: number | null;
}

export interface FullMatchListResponse {
  total: number;
  limit: number;
  offset: number;
  matches: FullMatchSummaryRow[];
}

export interface FullMatchListFilters {
  player?: string;
  surface?: string;
  year?: number;
  tourney_level?: string;
  limit?: number;
  offset?: number;
}

File: tennis-intelligence-platform/api/schemas/full_match_list.py
"""
schemas/full_match_list.py — Pydantic response models for the FULL-corpus Match
Explorer endpoint. Genuinely separate from schemas/match_list.py, which backs the
existing, MCP-limited endpoint.
"""

from __future__ import annotations

from pydantic import BaseModel


class FullMatchSummary(BaseModel):
    match_id: str
    has_replay_data: bool
    tournament: str
    year: int | None
    surface: str
    round: str
    winner: str
    loser: str
    final_score: str | None
    tournament_level: str | None
    best_of: int | None
    winner_elo: float | None
    loser_elo: float | None


class FullMatchListResponse(BaseModel):
    total: int
    limit: int
    offset: int
    matches: list[FullMatchSummary]